### PR TITLE
Setting web3 provider when web3 context is toggled

### PR DESCRIPTION
**Pull request by redsquirrel**
When a user chooses the "Web3 Provider" option, it doesn't actually set the provider unless the endpoint value is changed.

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -243,22 +243,28 @@ var run = function() {
...
 	$web3Toggle.on('change', executionContextChange );
 	$web3endpoint.on('change', function() {
-		var endpoint = $web3endpoint.val();
-		if (endpoint == 'ipc')
-			web3.setProvider(new web3.providers.IpcProvider());
-		else
-			web3.setProvider(new web3.providers.HttpProvider(endpoint));
-		compile();
+		setProviderFromEndpoint();
+		if (executionContext == 'web3') compile();
```
Comment (chriseth) [inline src/app.js]: Why only in this case?

Comment (redsquirrel) [inline src/app.js]: I'm making two assumptions here:
* the only possible values for `executionContext` are `web3` and `vm`
* the value of `endpoint` is meaningless while in `vm` context

Comment (chriseth) [inline src/app.js]: Oh now I see. We really need to refactor that code...

Comment (redsquirrel) [inline src/app.js]: challenge accepted! 🚑

**Pull request merged**